File: app/src/main/java/io/github/takusan23/kaisendon/APIJSONParse/CustomMenuJSONParse.java
package io.github.takusan23.kaisendon.APIJSONParse;

import org.json.JSONException;
import org.json.JSONObject;

public class CustomMenuJSONParse {
    String misskey = "";
    String name = "";
    String content = "";
    String instance = "";
    String access_token = "";
    String image_load = "";
    String dialog = "";
    String dark_mode = "";
    String position = "";
    String streaming = "";
    String subtitle = "";
    String image_url = "";
    String background_transparency = "";
    String background_screen_fit = "";
    String quick_profile = "";
    String toot_counter = "";
    String custom_emoji = "";
    String gif = "";
    String font = "";
    String one_hand = "";
    String misskey_username = "";
    String setting = "";

    public CustomMenuJSONParse(String json_data) {
        json_parse(json_data);
    }

    private void json_parse(String json_data) {
        try {
            JSONObject jsonObject = new JSONObject(json_data);
            name = jsonObject.getString("name");
            content = jsonObject.getString("content");
            instance = jsonObject.getString("instance");
            access_token = jsonObject.getString("access_token");
            image_load = jsonObject.getString("image_load");
            dialog = jsonObject.getString("dialog");
            dark_mode = jsonObject.getString("dark_mode");
            position = jsonObject.getString("position");
            streaming = jsonObject.getString("streaming");
            subtitle = jsonObject.getString("subtitle");
            image_url = jsonObject.getString("image_url");
            background_transparency = jsonObject.getString("background_transparency");
            background_screen_fit = jsonObject.getString("background_screen_fit");
            quick_profile = jsonObject.getString("quick_profile");
            toot_counter = jsonObject.getString("toot_counter");
            custom_emoji = jsonObject.getString("custom_emoji");
            gif = jsonObject.getString("gif");
            font = jsonObject.getString("font");
            one_hand = jsonObject.getString("one_hand");
            misskey = jsonObject.getString("misskey");
            misskey_username = jsonObject.getString("misskey_username");
            setting = jsonObject.getString("setting");
        } catch (JSONException e) {
            e.printStackTrace();
        }
    }

    public String getMisskey() {
        return misskey;
    }

    public String getName() {
        return name;
    }

    public String getContent() {
        return content;
    }

    public String getInstance() {
        return instance;
    }

    public String getAccess_token() {
        return access_token;
    }

    public String getImage_load() {
        return image_load;
    }

    public String getDialog() {
        return dialog;
    }

    public String getDark_mode() {
        return dark_mode;
    }

    public String getPosition() {
        return position;
    }

    public String getStreaming() {
        return streaming;
    }

    public String getSubtitle() {
        return subtitle;
    }

    public String getImage_url() {
        return image_url;
    }

    public String getBackground_transparency() {
        return background_transparency;
    }

    public String getBackground_screen_fit() {
        return background_screen_fit;
    }

    public String getQuick_profile() {
        return quick_profile;
    }

    public String getToot_counter() {
        return toot_counter;
    }

    public String getCustom_emoji() {
        return custom_emoji;
    }

    public String getGif() {
        return gif;
    }

    public String getFont() {
        return font;
    }

    public String getOne_hand() {
        return one_hand;
    }

    public String getMisskey_username() {
        return misskey_username;
    }

    public String getSetting() {
        return setting;
    }
}

File: app/src/main/java/io/github/takusan23/Kaisendon/APIJSONParse/CustomMenuJSONParse.kt
package io.github.takusan23.Kaisendon.APIJSONParse

import org.json.JSONException
import org.json.JSONObject

class CustomMenuJSONParse(json_data: String) {
    var misskey = ""
        internal set
    var name = ""
        internal set
    var content = ""
        internal set
    var instance = ""
        internal set
    var access_token = ""
        internal set
    var image_load = ""
        internal set
    var dialog = ""
        internal set
    var dark_mode = ""
        internal set
    var position = ""
        internal set
    var streaming = ""
        internal set
    var subtitle = ""
        internal set
    var image_url = ""
        internal set
    var background_transparency = ""
        internal set
    var background_screen_fit = ""
        internal set
    var quick_profile = ""
        internal set
    var toot_counter = ""
        internal set
    var custom_emoji = ""
        internal set
    var gif = ""
        internal set
    var font = ""
        internal set
    var one_hand = ""
        internal set
    var misskey_username = ""
        internal set
    var isReadOnly = ""
        internal set
    var setting = ""
        internal set
    var json_data = ""
    //テーマ機能
    var theme_data = false  //テーマ設定があるか
    var theme_darkmode = ""
    var theme_status_bar_color = ""
    var theme_nav_bar_color = ""
    var theme_tool_bar_color = ""
    var theme_background_color = ""
    var theme_toot_background_color = ""
    var theme_text_icon_color = ""

    init {
        json_parse(json_data)
    }

    private fun json_parse(json_data: String) {
        try {

            val jsonObject = JSONObject(json_data)
            this.json_data = json_data
            name = getStringJsonCheck(jsonObject, "name")
            content = getStringJsonCheck(jsonObject, "content")
            instance = getStringJsonCheck(jsonObject, "instance")
            access_token = getStringJsonCheck(jsonObject, "access_token")
            image_load = getStringJsonCheck(jsonObject, "image_load")
            dialog = getStringJsonCheck(jsonObject, "dialog")
            dark_mode = getStringJsonCheck(jsonObject, "dark_mode")
            position = getStringJsonCheck(jsonObject, "position")
            streaming = getStringJsonCheck(jsonObject, "streaming")
            subtitle = getStringJsonCheck(jsonObject, "subtitle")
            image_url = getStringJsonCheck(jsonObject, "image_url")
            background_transparency = getStringJsonCheck(jsonObject, "background_transparency")
            background_screen_fit = getStringJsonCheck(jsonObject, "background_screen_fit")
            quick_profile = getStringJsonCheck(jsonObject, "quick_profile")
            toot_counter = getStringJsonCheck(jsonObject, "toot_counter")
            custom_emoji = getStringJsonCheck(jsonObject, "custom_emoji")
            gif = getStringJsonCheck(jsonObject, "gif")
            font = getStringJsonCheck(jsonObject, "font")
            one_hand = getStringJsonCheck(jsonObject, "one_hand")
            misskey = getStringJsonCheck(jsonObject, "misskey")
            misskey_username = getStringJsonCheck(jsonObject, "misskey_username")
            setting = getStringJsonCheck(jsonObject, "setting")
            isReadOnly = getStringJsonCheckReturnText(jsonObject, "read_only", "false")
            //テーマ機能
            if (jsonObject.has("theme")) {
                theme_data = true
                val themeJsonObject = jsonObject.getJSONObject("theme")
                theme_darkmode = getStringJsonCheckReturnText(themeJsonObject, "theme_darkmode", "false")
                theme_status_bar_color = getStringJsonCheck(themeJsonObject, "theme_status_bar_color")
                theme_nav_bar_color = getStringJsonCheck(themeJsonObject, "theme_nav_bar_color")
                theme_tool_bar_color = getStringJsonCheck(themeJsonObject, "theme_tool_bar_color")
                theme_background_color = getStringJsonCheck(themeJsonObject, "theme_background_color")
                theme_toot_background_color = getStringJsonCheck(themeJsonObject, "theme_toot_background_color")
                theme_text_icon_color = getStringJsonCheck(themeJsonObject, "theme_text_icon_color")
            }
        } catch (e: JSONException) {
            e.printStackTrace()
        }
    }

    fun getStringJsonCheck(jsonObject: JSONObject, string: String): String {
        if (jsonObject.has(string)) {
            return jsonObject.getString(string)
        } else {
            return ""
        }
    }

    fun getStringJsonCheckReturnText(jsonObject: JSONObject, string: String, error: String): String {
        if (jsonObject.has(string)) {
            return jsonObject.getString(string)
        } else {
            return error
        }
    }

}
